.pre-commit-config.yaml

```diff
@@ -0,0 +1,10 @@
+repos:
```

Comment:
Suggestion:
Por que não algo como?

```yml
repos:
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.5.0
    hooks:
      - id: check-yaml
        args:
          - --unsafe
      - id: end-of-file-fixer
      - id: trailing-whitespace
  - repo: https://github.com/pdm-project/pdm
    rev: 2.17.1
    hooks:
      - id: pdm-sync
  - repo: local
    hooks:
      - id: pre-commit-hook
        name: pdm-check
        entry: pdm run check
        language: system
        always_run: true
        files: \.(py)$
```


Sugiro adicionar o `black --check` como comando para o `pdm check` assim verifica tb se formatação esta ok.

Reply:
Então.. vamos lá.

Removi a referência do pre-commit-hooks padrão da lib pois eles traz opinião sendo que a gente pode querer customizar no nosso gosto, tendo em vista que a gente até já tem um `pdm format` que toma conta dessa parte.

O pdm check ele só checa coisas, ele não faz nada. A ideia do pre-commit hook é justamente detectar erros e tentar já corrigi-los se for corrigível automaticamente, que é o caso do `pdm format` e o `pdm install`, que atualiza suas deps e a formatação do teu código já para você sem precisar fazer manualmente toda vez, que seu .yaml não contempla.

O único que não se corrige sozinho é rodar os testes com `pdm tests`, onde ele só vai acusar os erros que tiver, que no caso do seu .yaml nem contempla.

O script de format e tests está no pyproject.toml, só checar lá. Já são scripts existentes que o Manoel escreveu.

Reply:
Man, o `pdm run check` roda os testes sim, é o que esta definido no `pyproject.toml`.
E com relação a todo o resto, seria somente o caso de criar uma nova entrada no pdm q faz tudo isso q queremos.
Minha sugestão foi + para não ter um script externo sendo q o pdm meio q já faz isso.